mongo/description/selector_test.go

```diff
@@ -230,6 +230,67 @@ func TestSelector_Sharded(t *testing.T) {
 	require.Equal([]Server{s}, result)
 }
 
+func BenchmarkLatencySelector(b *testing.B) {
+	bench := func(b *testing.B, serversHook func(servers []Server)) {
```

Comment:
Instead of using a closure and constructing sub-benchmarks for each case, can we iterate over the benchmark cases? Something like this:

```go
	for _, bcase := range []struct {
		name        string
		serversHook func(servers []description.Server)
	}{
		{
			"AllFit",
			func(servers []description.Server) {},
		},
		{
			"AllButOneFit",
			func(servers []description.Server) {
				servers[0].AverageRTT = 2 * time.Second
			},
		},
		// Continue test cases
	} {
		bcase := bcase

		b.Run(bcase.name, func(b *testing.B) {
			// your benchmark logic
		})
	}
```

Reply:
Changed benchmark this way